pkg/test/expectations/expectations.go

```diff
@@ -599,3 +600,33 @@ func ExpectMachinesWithOffset(offset int, ctx context.Context, c client.Client)
 		return &m
 	})
 }
+
+func ExpectMakeNodesMachinesInitializedAndStateUpdated(ctx context.Context, c client.Client, nodeStateController, machineStateController controller.Controller, nodes []*v1.Node, machines []*v1alpha5.Machine) {
```

Comment:
Should we just break up these functions to be? 

`ExpectNodesInitializedAndStateUpdated`
`ExpectMachinesInitializedAndStateUpdated`
`ExpectNodeClaimsInitializedAndStateUpdated`

I'm wondering if they're getting far too wordy and it might be best to just decouple them

Reply:
Yeah, personally, I think this is acceptable, and we merged these functions together since the deprovisioning code calls the ExpectNodes... and ExpectMachines... call a lot. 

I renamed it to be ...NodeNodeClaims... because of the fact that we have to replicate the Provisioner/NodePool Machine/NodeClaim logic across the codebase. When we remove the instances of Provisioners/Machines from the codebase, I think we can rename this back to ExpectMakeInitializedAndStateUpdated(). Thoughts?

Reply:
Sure, I might opt for throwing the word `And` in here so that it's a little more readable, but I'm fine whatever name you decide:

```go
ExpectMakeNodesAndMachinesInitializedAndStateUpdated
ExpectMakeNodeClaimsAndMachineInitializedAndStateUpdated
```